e2e/helm_charts_test.go

```diff
@@ -0,0 +1,521 @@
+	}
+
+	return ""
+}
+
+func (ts *HelmChartTestSuite) install(chart string) error {
+	ts.Name = chart
+
+	elasticChart := "elastic/" + ts.Name
+
+	flags := []string{}
+	if chart == "elasticsearch" {
+		// Rancher Local Path Provisioner and local-path storage class for Elasticsearch volumes
+		args := []string{
+			"apply", "-f", "https://raw.githubusercontent.com/rancher/local-path-provisioner/master/deploy/local-path-storage.yaml",
+		}
+
+		_, err := shell.Execute(".", "kubectl", args...)
+		if err != nil {
+			log.Errorf("Could not apply Rancher Local Path Provisioner: %v", err)
+			return err
+		}
+		log.WithFields(log.Fields{
+			"name": ts.Name,
+		}).Debug("Rancher Local Path Provisioner and local-path storage class for Elasticsearch volumes installed")
+
+		// workaround to use Rancher's local-path storage class for Elasticsearch volumes
+		flags = []string{"--wait", "--timeout=900", "--values", "https://raw.githubusercontent.com/elastic/helm-charts/master/elasticsearch/examples/kubernetes-kind/values.yaml"}
+	}
+
+	return helm.InstallChart(ts.Name, elasticChart, ts.Version, flags)
+}
+
+func (ts *HelmChartTestSuite) installRuntimeDependencies(dependencies ...string) {
+	for _, dependency := range dependencies {
+		// Install Elasticsearch
+		err := ts.install(dependency)
+		if err != nil {
+			log.Fatalf("Could not install %s as runtime dependency. Aborting: %v", dependency, err)
+		}
+	}
+}
+
+func (ts *HelmChartTestSuite) podsManagedByDaemonSet() error {
+	output, err := kubectl.Run("get", "daemonset", "--namespace=default", "-l", "app="+ts.Name+"-"+ts.Name, "-o", "jsonpath='{.items[0].metadata.labels.chart}'")
+	if err != nil {
+		return err
+	}
+	if output != ts.getFullName() {
+		return fmt.Errorf("There is no DaemonSet for the %s chart. Expected: %s, Actual: %s", ts.Name, ts.getFullName(), output)
+	}
+
+	log.WithFields(log.Fields{
+		"output": output,
+		"name":   ts.Name,
+	}).Debug("A pod will be deployed on each node of the cluster by a DaemonSet")
+
+	return nil
+}
+
+func (ts *HelmChartTestSuite) resourceWillManageAdditionalPodsForMetricsets(resource string) error {
+	lowerResource := strings.ToLower(resource)
+
+	output, err := kubectl.Run("get", lowerResource, ts.getResourceName(resource), "-o", "jsonpath='{.metadata.labels.chart}'")
+	if err != nil {
+		return err
+	}
+	if output != ts.getFullName() {
+		return fmt.Errorf("There is no %s for the %s chart. Expected: %s, Actual: %s", resource, ts.Name, ts.getFullName(), output)
+	}
+
+	log.WithFields(log.Fields{
+		"output": output,
+		"name":   ts.Name,
+	}).Debug("A " + resource + " will manage additional pods for metricsets querying internal service")
+
+	return nil
+}
+
+func (ts *HelmChartTestSuite) strategyCanBeUsedDuringUpdates(strategy string) error {
+	output, err := kubectl.Run("get", "ds/"+ts.Name+"-"+ts.Name, "-o", `go-template='{{.spec.updateStrategy.type}}'`)
+	if err != nil {
+		return err
+	}
+
+	output = strings.ReplaceAll(output, "'", "")
```

Comment:
the output does not have `'` characters

Reply:
Solved! The go-template I was using included extra quotes, which I removed